src/plugins/legacy/polygonRoi/polygonEventFilter.cpp

```diff
@@ -306,9 +315,15 @@ bool polygonEventFilter::leftButtonBehaviour(medAbstractView *view, QMouseEvent
     }
     case CURSORSTATE::CS_REPULSOR:
     {
+#if QT_VERSION > QT_VERSION_CHECK(5, 10, 0)
+        int devicePixelRatio = QGuiApplication::screenAt(mouseEvent->globalPos())->devicePixelRatio();
+#else
+        int screenNumber = QApplication::desktop()->screenNumber(mouseEvent->globalPos());
+        int devicePixelRatio = QGuiApplication::screens().at(screenNumber)->devicePixelRatio();
+#endif
         double mousePos[2];
-        mousePos[0] = mouseEvent->x()*QGuiApplication::screenAt(mouseEvent->globalPos())->devicePixelRatio();
-        mousePos[1] = (currentView->viewWidget()->height()-mouseEvent->y()-1)*QGuiApplication::screenAt(mouseEvent->globalPos())->devicePixelRatio();
+        mousePos[0] = mouseEvent->x()*devicePixelRatio;
```

Comment:
by the way @juliencastelneau my compiler says that mousePos is not used in this function

Reply:
yes, you are right. We could remove lines